### DISCUSSION TITLE
How should the quickstart document deal with API connections and secrets?

### DISCUSSION THREAD

**Discussion by BurtHarris** [microsoft/promptflow#3631] [General]
Reading the Prompt Flow documents in the order presented leaves the reader rather clueless about setting up API connections and API keys.  
   
Following the documented quick start steps leads to: `pf.flow.test failed with UserErrorException: RequiredEnvironmentVariablesNotSetError: Execution failure in 'chat': (RequiredEnvironmentVariablesNotSetError) Required environment variables ['AZURE_OPENAI_ENDPOINT', 'AZURE_OPENAI_API_KEY'] to build AzureOpenAIConnection not set.`

The sample code in the quick start written to load env variables from a .env file,   Is that the plan?   If so, he "Quick Start" guide needs guidance configuring API connections used in the example.

But it appears that Prompt Flow has provisions for making connections objects using YAML.   I don't understand, is this a mitigation for the security risks of storing API keys in files?   I'm not seeing the big picture in the documentation.   Does it integrate with Azure Key Vault or something?

**Comment by brynn-code** [comment]
Hi @BurtHarris, thanks for reaching us. We support both environment variable and connection in this case, but for quick start, we'd like to introduce concept as less as possible, so that's a minor flow to let user know what is a flow, and able to run it simply. 
For the connection related question, you could find more examples here: https://microsoft.github.io/promptflow/tutorials/quickstart.html
